WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -224,6 +269,10 @@ private async Task CreateBlameRoundAsync(Round round, CancellationToken cancella
 			var feeRate = (await Rpc.EstimateSmartFeeAsync((int)Config.ConfirmationTarget, EstimateSmartFeeMode.Conservative, simulateIfRegTest: true).ConfigureAwait(false)).FeeRate;
 			RoundParameters parameters = new(Config, Network, Random, feeRate, blameOf: round);
 			Round blameRound = new(parameters);
+			if (!RoundsById.TryAdd(blameRound.Id, blameRound))
+			{
+				throw new InvalidOperationException();
```

Comment:
it should use `Add` instead.

But I don't think this should use a normal dictionary, unless we use some sort of shared lock abstraction that would allow multiple simultaneous readers. The rounds collection changes very rarely, but is used by every single request so if that's protected by a lock it will linearize everything just like `AsyncLock` does now (although perhaps to a smaller scope).

Reply:
For what its worth I think we could use a `ReaderWriterLock{,Slim}` instead of a concurrent dictionary because all of the `await` calls performed by `Arena` (or later `Round`) should be outside of the critical section. I initially went with a concurrent dictionary initially because `AsyncLock` doesn't have an RW equivalent,  but by the time I was done I think it became apparent that the critical sections are pretty small and and don't do anything complicated, so it's definitely possible.